rollbackFuture: Self-suppression not permitted

The following trace is occasionally observed on our sharded Linux workers on revision 4b47f4aa73d68f2d36083d137a8af0ac0b5e662d:

```
error creating exec dir for shard/operations/c8a03436-710c-45f6-b00d-3fe6c96f95c1
build.buildfarm.worker.shard.CFCExecFileSystem: /data/shard/operations/c8a03436-710c-45f6-b00d-3fe6c96f95c1: 1 exceptions
	at build.buildfarm.worker.shard.CFCExecFileSystem.checkExecErrors(CFCExecFileSystem.java:306)
	at build.buildfarm.worker.shard.CFCExecFileSystem.createExecDir(CFCExecFileSystem.java:370)
	at build.buildfarm.worker.shard.ShardWorkerContext.createExecDir(ShardWorkerContext.java:778)
	at build.buildfarm.worker.InputFetcher.fetchPolled(InputFetcher.java:179)
	at build.buildfarm.worker.InputFetcher.runInterruptibly(InputFetcher.java:85)
	at build.buildfarm.worker.InputFetcher.run(InputFetcher.java:269)
	at java.base/java.lang.Thread.run(Thread.java:829)
	Suppressed: java.lang.IllegalArgumentException: Self-suppression not permitted
		at java.base/java.lang.Throwable.addSuppressed(Throwable.java:1054)
		at build.buildfarm.cas.cfc.CASFileCache.lambda2(CASFileCache.java:2233)
		at com.google.common.util.concurrent.AbstractCatchingFuture.doFallback(AbstractCatchingFuture.java:192)
		at com.google.common.util.concurrent.AbstractCatchingFuture.doFallback(AbstractCatchingFuture.java:179)
		at com.google.common.util.concurrent.AbstractCatchingFuture.run(AbstractCatchingFuture.java:124)
		at java.base/java.util.concurrent.ForkJoinTask.exec(ForkJoinTask.java:1426)
		at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
		at java.base/java.util.concurrent.ForkJoinPool.topLevelExec(ForkJoinPool.java:1020)
		at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
		at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
		at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
	Caused by: java.nio.file.NoSuchFileException: /data/cache/4e/54/4e54503a0b3d298eb9942e4e53f31196027322642eff42b75eceb9ec1008cf74_dir/bin/jcmd
		at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
		at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
		at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
		at java.base/sun.nio.fs.UnixFileAttributeViews.readAttributes(UnixFileAttributeViews.java:55)
		at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149)
		at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
		at java.base/java.nio.file.Files.readAttributes(Files.java:1764)
		at java.base/java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:219)
		at java.base/java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
		at java.base/java.nio.file.FileTreeWalker.next(FileTreeWalker.java:373)
		at java.base/java.nio.file.Files.walkFileTree(Files.java:2761)
		at java.base/java.nio.file.Files.walkFileTree(Files.java:2797)
		at build.buildfarm.common.io.Directories.remove(Directories.java:112)
		at build.buildfarm.cas.cfc.CASFileCache.fetchDirectory(CASFileCache.java:1961)
		at build.buildfarm.cas.cfc.CASFileCache.lambda0(CASFileCache.java:2156)
		at com.google.common.util.concurrent.AbstractTransformFuture.doTransform(AbstractTransformFuture.java:213)
		at com.google.common.util.concurrent.AbstractTransformFuture.doTransform(AbstractTransformFuture.java:202)
		at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:118)
		... 6 common frames omitted
```

I could not find any instance of 'error while' anywhere in the workers' logs. Around that time, another unrelated worker displayed:

```
error creating exec dir for shard/operations/d7e51406-b4ad-4ae2-8da2-6caaef5adb30
build.buildfarm.worker.shard.CFCExecFileSystem: /data/shard/operations/d7e51406-b4ad-4ae2-8da2-6caaef5adb30: 1 exceptions
	at build.buildfarm.worker.shard.CFCExecFileSystem.checkExecErrors(CFCExecFileSystem.java:306)
	at build.buildfarm.worker.shard.CFCExecFileSystem.createExecDir(CFCExecFileSystem.java:370)
	at build.buildfarm.worker.shard.ShardWorkerContext.createExecDir(ShardWorkerContext.java:778)
	at build.buildfarm.worker.InputFetcher.fetchPolled(InputFetcher.java:179)
	at build.buildfarm.worker.InputFetcher.runInterruptibly(InputFetcher.java:85)
	at build.buildfarm.worker.InputFetcher.run(InputFetcher.java:269)
	at java.base/java.lang.Thread.run(Thread.java:829)
	Suppressed: java.nio.file.NoSuchFileException: /data/cache/4e/54/4e54503a0b3d298eb9942e4e53f31196027322642eff42b75eceb9ec1008cf74_dir/lib
		at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
		at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
		at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
		at java.base/sun.nio.fs.UnixFileAttributeViews.setMode(UnixFileAttributeViews.java:254)
		at java.base/sun.nio.fs.UnixFileAttributeViews.setPermissions(UnixFileAttributeViews.java:276)
		at java.base/java.nio.file.Files.setPosixFilePermissions(Files.java:2080)
		at build.buildfarm.common.io.Directories.makeWritable(Directories.java:56)
		at build.buildfarm.common.io.Directories.access-bash00(Directories.java:42)
		at build.buildfarm.common.io.Directories.preVisitDirectory(Directories.java:119)
		at build.buildfarm.common.io.Directories.preVisitDirectory(Directories.java:114)
		at java.base/java.nio.file.Files.walkFileTree(Files.java:2732)
		at java.base/java.nio.file.Files.walkFileTree(Files.java:2797)
		at build.buildfarm.common.io.Directories.remove(Directories.java:112)
		at build.buildfarm.cas.cfc.CASFileCache.fetchDirectory(CASFileCache.java:1961)
		at build.buildfarm.cas.cfc.CASFileCache.lambda0(CASFileCache.java:2156)
		at com.google.common.util.concurrent.AbstractTransformFuture.doTransform(AbstractTransformFuture.java:213)
		at com.google.common.util.concurrent.AbstractTransformFuture.doTransform(AbstractTransformFuture.java:202)
		at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:118)
		at java.base/java.util.concurrent.ForkJoinTask.exec(ForkJoinTask.java:1426)
		at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
		at java.base/java.util.concurrent.ForkJoinPool.topLevelExec(ForkJoinPool.java:1020)
		at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
		at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
		at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

and about 15 seconds later, a third worker displayed:

```
error uploading outputs for shard/operations/2a3bdb73-262f-4b3a-b911-8f8fa2b519c8
io.grpc.StatusRuntimeException: RESOURCE_EXHAUSTED
	at build.buildfarm.worker.shard.Worker.insertStream(Worker.java:167)
	at build.buildfarm.worker.shard.Worker.insertBlob(Worker.java:145)
	at build.buildfarm.worker.shard.ShardWorkerContext.insertBlob(ShardWorkerContext.java:503)
	at build.buildfarm.worker.shard.ShardWorkerContext.updateActionResultStdOutputs(ShardWorkerContext.java:518)
	at build.buildfarm.worker.shard.ShardWorkerContext.uploadOutputs(ShardWorkerContext.java:740)
	at build.buildfarm.worker.ReportResultStage.reportPolled(ReportResultStage.java:97)
	at build.buildfarm.worker.ReportResultStage.tick(ReportResultStage.java:80)
	at build.buildfarm.worker.PipelineStage.iterate(PipelineStage.java:98)
	at build.buildfarm.worker.PipelineStage.runInterruptible(PipelineStage.java:44)
	at build.buildfarm.worker.PipelineStage.run(PipelineStage.java:51)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: io.grpc.StatusRuntimeException: RESOURCE_EXHAUSTED
	at io.grpc.Status.asRuntimeException(Status.java:526)
	... 11 common frames omitted
Caused by: java.io.IOException: java.lang.InterruptedException
	at build.buildfarm.cas.cfc.CASFileCache.newOutput(CASFileCache.java:984)
	at build.buildfarm.cas.cfc.CASFileCache.createUniqueWriteOutput(CASFileCache.java:938)
	at build.buildfarm.cas.cfc.CASFileCache.getOutput(CASFileCache.java:894)
	at build.buildfarm.worker.shard.Worker.insertStream(Worker.java:161)
	... 10 common frames omitted
Caused by: java.lang.InterruptedException: null
	at build.buildfarm.cas.cfc.CASFileCache.charge(CASFileCache.java:2565)
	at build.buildfarm.cas.cfc.CASFileCache.putOrReferenceGuarded(CASFileCache.java:2585)
	at build.buildfarm.cas.cfc.CASFileCache.putOrReference(CASFileCache.java:2478)
	at build.buildfarm.cas.cfc.CASFileCache.putImpl(CASFileCache.java:2376)
	at build.buildfarm.cas.cfc.CASFileCache.newOutput(CASFileCache.java:974)
	... 13 common frames omitted
```

We run sharded _servers_ behind a load balancer: could retries cause the concurrent operations being observed?

--- a/src/main/java/build/buildfarm/cas/cfc/CASFileCache.java
+++ b/src/main/java/build/buildfarm/cas/cfc/CASFileCache.java
@@ -2230,7 +2230,7 @@ public abstract class CASFileCache implements ContentAddressableStorage {
                     Level.SEVERE,
                     "error during directory removal after fetch failure of " + path,
                     removeException);
-                e.addSuppressed(e);
+                e.addSuppressed(removeException);
               }
               return immediateFailedFuture(e);
             },

